drivers/sensor/shtcx/shtcx.c

```diff
@@ -0,0 +1,279 @@
+	return i2c_write(shtcx_i2c_bus(dev), tx_buf, sizeof(tx_buf),
+			 shtcx_i2c_address(dev));
+}
+
+static int shtcx_read_words(const struct device *dev, uint16_t cmd, uint16_t *data,
+		     uint16_t num_words, uint16_t max_duration_us)
+{
+	const struct shtcx_config *cfg = dev->config;
+	int status = 0;
+	uint32_t raw_len = num_words * (SHTCX_WORD_LEN + SHTCX_CRC8_LEN);
+	uint16_t temp16;
+	uint8_t rx_buf[SHTCX_MAX_READ_LEN];
+	int dst = 0;
+
+	status = shtcx_write_command(dev, cmd);
+	if (status != 0) {
+		LOG_DBG("Failed to initiate read");
+		return -EIO;
+	}
+
+	if (!cfg->clock_stretching) {
+		k_sleep(K_USEC(max_duration_us));
+	}
+
+	status = i2c_read(shtcx_i2c_bus(dev), rx_buf, raw_len,
+			  shtcx_i2c_address(dev));
+	if (status != 0) {
+		LOG_DBG("Failed to read data");
+		return -EIO;
+	}
+
+	for (int i = 0; i < raw_len; i += (SHTCX_WORD_LEN + SHTCX_CRC8_LEN)) {
+		temp16 = sys_get_be16(&rx_buf[i]);
+		if (shtcx_compute_crc(temp16) != rx_buf[i+2]) {
+			LOG_DBG("invalid received invalid crc");
+			return -EIO;
+		}
+
+		data[dst++] = temp16;
+	}
+
+	return 0;
+}
+
+static int shtcx_sleep(const struct device *dev)
+{
+	if (shtcx_write_command(dev, SHTCX_CMD_SLEEP) < 0) {
+		return -EIO;
+	}
+	return 0;
+}
+
+static int shtcx_wakeup(const struct device *dev)
+{
+	if (shtcx_write_command(dev, SHTCX_CMD_WAKEUP)) {
+		return -EIO;
+	}
+	k_sleep(K_USEC(100));
+	return 0;
+}
+
+static int shtcx_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct shtcx_data *data = dev->data;
+	const struct shtcx_config *cfg = dev->config;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (cfg->chip == SHTC3) {
+		if (shtcx_wakeup(dev)) {
+			return -EIO;
+		}
+	}
+
+	if (shtcx_read_words(dev,
+			     measure_cmd[cfg->measure_mode][cfg->clock_stretching],
+			     (uint16_t *)&data->sample, 2,
+			     measure_wait_us[cfg->chip][cfg->measure_mode]) < 0) {
+		LOG_DBG("Failed read measurements!");
+		return -EIO;
+	}
+
+	if (cfg->chip == SHTC3) {
+		if (shtcx_sleep(dev)) {
+			LOG_DBG("Failed to initiate sleep");
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int shtcx_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct shtcx_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		shtcx_temperature_from_raw(data->sample.temp, val);
+	} else if (chan == SENSOR_CHAN_HUMIDITY) {
+		shtcx_humidity_from_raw(data->sample.temp, val);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api shtcx_driver_api = {
+	.sample_fetch = shtcx_sample_fetch,
+	.channel_get = shtcx_channel_get,
+};
+
+static int shtcx_init(const struct device *dev)
+{
+	const struct shtcx_config *cfg = dev->config;
+	uint16_t product_id;
+
+	if (device_is_ready(cfg->bus) ==  0) {
+		LOG_DBG("i2c bus is not ready");
+		return -ENODEV;
+	}
+
+	if (!cfg->base_address) {
+		LOG_ERR("No I2C address");
+		return -EINVAL;
+	}
```

Comment:
This does not seem needed. If someone puts 0 in the devicetree `reg` for this node, that is a simple misconfiguration, no? What am I missing?

Reply:
removed